catalyst-gateway/bin/src/service/api/cardano/rbac/registrations_get/key_data.rs

```diff
@@ -0,0 +1,146 @@
+//! A role data key information.
+
+use std::collections::HashMap;
+
+use anyhow::Context;
+use c509_certificate::c509::C509;
+use cardano_blockchain_types::Point;
+use chrono::{DateTime, Utc};
+use ed25519_dalek::VerifyingKey;
+use minicbor::{encode::Encode, Encoder};
+use poem_openapi::{types::Example, Object};
+use rbac_registration::{
+    cardano::cip509::{KeyLocalRef, LocalRefInt, PointData},
+    registration::cardano::RegistrationChain,
+};
+use x509_cert::{certificate::Certificate as X509Certificate, der::Encode as _};
+
+use crate::service::{
+    api::cardano::rbac::registrations_get::binary_data::HexEncodedBinaryData,
+    common::types::generic::{
+        boolean::BooleanFlag, date_time::DateTime as ServiceDateTime,
+        ed25519_public_key::Ed25519HexEncodedPublicKey,
+    },
+};
+
+/// A role data key information.
+///
+/// Only one of the `pub_key`, `x509_cert` and `c509_cert` fields can be present at the
+/// same time.
+#[derive(Object, Debug, Clone)]
+#[oai(example)]
+pub struct KeyData {
+    /// Indicates if the data is persistent or volatile.
+    is_persistent: BooleanFlag,
+    /// A time when the data was added.
+    time: ServiceDateTime,
+    /// A hex encoded X509 certificate.
```

Comment:
I am not sure this is a great way to return the data.
these fields are mutually exclusive, and the schema does not represent that.
It makes the schema confusing.
Maybe it should just be something like:
```
   key_type: <enum> "pubkey", "x509", c509"
   key_value: HexEncodedBinaryData
```
public keys are encoded the same.
Also, maybe we should use base64 encoding for this data, as thats the typical way to embed binary data inside json.

Reply:
Fixed.